From f2865e5deda545e8f090abfd8d884bf502a2de6d Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:25:33 -0400
Subject: [md] raid5: For stripe with R5_ReadNoMerge, we replace REQ_FLUSH with REQ_NOMERGE

Message-id: <1409563584-9000-32-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 89969
O-Subject: [RHEL7.1 PATCH 31/82] md/raid5: For stripe with R5_ReadNoMerge, we replace REQ_FLUSH with REQ_NOMERGE.
Bugzilla: 1085530
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: majianpeng <majianpeng@gmail.com>

For R5_ReadNoMerge,it mean this bio can't merge with other bios or
request.It used REQ_FLUSH to achieve this. But REQ_NOMERGE can do the
same work.

Signed-off-by: Jianpeng Ma <majianpeng@gmail.com>
Signed-off-by: NeilBrown <neilb@suse.de>
(cherry picked from commit e59aa23f4c0c61b4baa5d535a2071d6666e83502)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/raid5.c b/drivers/md/raid5.c
index a3100c1..5730f43 100644
--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -860,7 +860,7 @@ static void ops_run_io(struct stripe_head *sh, struct stripe_head_state *s)
     bi->bi_sector = (sh->sector
        + rdev->data_offset);
    if (test_bit(R5_ReadNoMerge, &sh->dev[i].flags))
-    bi->bi_rw |= REQ_FLUSH;
+    bi->bi_rw |= REQ_NOMERGE;
 
    bi->bi_vcnt = 1;
    bi->bi_io_vec[0].bv_len = STRIPE_SIZE;
-- 
1.7.1